### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by DanielVoogsgerd**
A collection of several clippy fixes.

This also denies the clippy suspicious code lint collection by default. It will deny lints like the mutable key types.

This might seem counterintuitive at first, but even though that code was sound, the comment that explicitly allows that code is good documentation in my opinion. It does indeed look suspicious.

TODO:
 - [x] Fix mutable key types
 - [x] Fix complicated types

### CODE REVIEW

**Comment by DanielVoogsgerd** [comment]
@Lut99 There are still two lints that warn on two complicated types. It has to do with workflow snippets in Brane-plr. I do agree with Clippy that it is not very intuitive what this type means.

Could you either fix it, or tell me how it should be called or structured, so we can make this better.

**Comment by DanielVoogsgerd** [comment]
@Lut99 Bump

File: brane-ast/src/traversals/print/ast_unresolved.rs [hunk 1]
```diff
@@ -306,7 +314,7 @@ pub fn pass_edges(
         },
     }
 
-    // DOne
```
Comment (Lut99) [inline brane-ast/src/traversals/print/ast_unresolved.rs:309]: Noooo, my comical bad capitalization :P

**Review by Lut99 — APPROVED** [review]

**Comment by Lut99** [comment]
Alright, let me have a look at the large type in `brane-plr`

**Comment by Lut99** [comment]
Hmm, I'm also getting a ton of "items after test module" warnings. Are we doing anything with that?

**Comment by Lut99** [comment]
> Could you either fix it, or tell me how it should be called or structured, so we can make this better.

Check it now!

**Pull request merged**